mlir/include/mlir/Dialect/AMDGPU/IR/AMDGPU.td

```diff
@@ -765,4 +771,37 @@ def AMDGPU_WMMAOp :
   let hasVerifier = 1;
 }
 
+def GlobalLoadMemRefType : MemRefOf<[GlobalLoadTypes]>;
+
+def AMDGPU_GlobalLoadLDSOp :
+    AMDGPU_Op<"global_load", [SameVariadicOperandSize]>,
+    Arguments<(ins
```

Comment:
This could be copying either one element at a time or be a gather of vectors.

Add a `TypeAttr:$copyType` or some such thing so that we know if it's going to gather `f16` or `vector<2xf16>` from the global memref